src/Umbraco.Web.BackOffice/Filters/EnsureUserPermissionForMediaAttribute.cs

```diff
@@ -0,0 +1,181 @@
+                }
+                else if (parameterValue is GuidUdi)
+                {
+                    guidId = ((GuidUdi) parameterValue).Guid;
+                }
+
+                if (guidId != Guid.Empty)
+                {
+                    var found = _entityService.GetId(guidId, UmbracoObjectTypes.Media);
+                    if (found)
+                        return found.Result;
+                }
+
+                throw new InvalidOperationException("The id type: " + parameterValue.GetType() +
+                                                    " is not a supported id");
+            }
+
+            public void OnActionExecuting(ActionExecutingContext context)
+            {
+                if (_webSecurity.CurrentUser == null)
+                {
+                    throw new HttpResponseException(System.Net.HttpStatusCode.Unauthorized);
+                }
+
+                int nodeId;
+                if (_nodeId.HasValue == false)
+                {
+                    var parts = _paramName.Split(new[] { '.' }, StringSplitOptions.RemoveEmptyEntries);
+
+                    if (context.ActionArguments[parts[0]] == null)
+                    {
+                        throw new InvalidOperationException("No argument found for the current action with the name: " +
+                                                            _paramName);
+                    }
+
+                    if (parts.Length == 1)
+                    {
+                        nodeId = GetNodeIdFromParameter(context.ActionArguments[parts[0]]);
+                    }
+                    else
+                    {
+                        //now we need to see if we can get the property of whatever object it is
+                        var pType = context.ActionArguments[parts[0]].GetType();
+                        var prop = pType.GetProperty(parts[1]);
+                        if (prop == null)
+                        {
+                            throw new InvalidOperationException(
+                                "No argument found for the current action with the name: " + _paramName);
+                        }
+
+                        nodeId = GetNodeIdFromParameter(prop.GetValue(context.ActionArguments[parts[0]]));
+                    }
+                }
+                else
+                {
+                    nodeId = _nodeId.Value;
+                }
+
+                if (MediaController.CheckPermissions(
+                    context.HttpContext.Items,
+                    _webSecurity.CurrentUser,
+                    _mediaService,
+                    _entityService,
+                    nodeId))
+                {
+
+                }
+                else
+                {
+                    throw new HttpResponseException(System.Net.HttpStatusCode.Unauthorized);
+                }
+            }
+
+            public void OnActionExecuted(ActionExecutedContext context)
+            {
+
```

Comment:
I can see this has shuffled around a little bit, so there are now some empty methods and we aren't needing to call the base method(s) here since we derive from ActionFilterAttribute. Just checking this is a conscious decision not to call it

Reply:
The reason is that we now implement `IActionFilter` which requires us to have both methods, even that we only use one.